test/integration/idle_timeout_integration_test.cc

```diff
@@ -181,5 +190,67 @@ TEST_P(IdleTimeoutIntegrationTest, PerStreamIdleTimeoutRequestAndResponse) {
   testRouterRequestAndResponseWithBody(1024, 1024, false, nullptr);
 }
 
+TEST_P(IdleTimeoutIntegrationTest, RequestTimeoutDisarmsOnRequestResponse) {
```

Comment:
This looks like it's testing more that the timeout doesn't *fire*, and the reason it doesn't fire is deep in the code.  If I'm right, reasonable to switch the name?

Reply:
Same comment below--I may be missing something, but the "Disarms" tests look like they're "DoesntFire" tests.